<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>encapp</class>
 <widget class="QDialog" name="encapp">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>604</width>
    <height>528</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Secret Exchange</string>
  </property>
  <widget class="QDialogButtonBox" name="buttonBox">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>10</y>
     <width>81</width>
     <height>241</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
   <property name="standardButtons">
    <set>QDialogButtonBox::Close</set>
   </property>
  </widget>
  <widget class="QWidget" name="formLayoutWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>491</width>
     <height>326</height>
    </rect>
   </property>
   <layout class="QFormLayout" name="formLayout">
    <property name="fieldGrowthPolicy">
     <enum>QFormLayout::ExpandingFieldsGrow</enum>
    </property>
    <item row="0" column="0">
     <widget class="QLabel" name="primeLabel">
      <property name="text">
       <string>Prime</string>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLineEdit" name="primeLineEdit"/>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="generatorLabel">
      <property name="text">
       <string>Generator</string>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLineEdit" name="generatorLineEdit"/>
    </item>
    <item row="2" column="0">
     <widget class="QLabel" name="aGALabel">
      <property name="text">
       <string>A = g^a</string>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QLineEdit" name="aGALineEdit"/>
    </item>
    <item row="3" column="0">
     <widget class="QLabel" name="bGBLabel">
      <property name="text">
       <string>B = g^b</string>
      </property>
     </widget>
    </item>
    <item row="3" column="1">
     <widget class="QLineEdit" name="bGBLineEdit"/>
    </item>
    <item row="4" column="0">
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="5" column="0">
     <widget class="QLabel" name="aliceSSecretALabel">
      <property name="text">
       <string>Alice's secret (a)</string>
      </property>
     </widget>
    </item>
    <item row="5" column="1">
     <widget class="QLineEdit" name="aliceSSecretALineEdit"/>
    </item>
    <item row="6" column="0">
     <widget class="QLabel" name="bobSSecretBLabel">
      <property name="text">
       <string>Bob's secret (b)</string>
      </property>
     </widget>
    </item>
    <item row="6" column="1">
     <widget class="QLineEdit" name="bobSSecretBLineEdit"/>
    </item>
    <item row="7" column="1">
     <widget class="Line" name="line_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item row="8" column="0">
     <widget class="QLabel" name="sharedSecretLabel">
      <property name="text">
       <string>Shared Secret</string>
      </property>
     </widget>
    </item>
    <item row="8" column="1">
     <widget class="QLineEdit" name="sharedSecretLineEdit"/>
    </item>
    <item row="11" column="0">
     <widget class="QLabel" name="inputLabel">
      <property name="text">
       <string>Input</string>
      </property>
     </widget>
    </item>
    <item row="11" column="1">
     <widget class="QLineEdit" name="inputLineEdit"/>
    </item>
    <item row="12" column="0">
     <widget class="QLabel" name="outputLabel">
      <property name="text">
       <string>Output</string>
      </property>
     </widget>
    </item>
    <item row="12" column="1">
     <widget class="QLineEdit" name="outputLineEdit"/>
    </item>
    <item row="9" column="0">
     <widget class="QLabel" name="md5label">
      <property name="text">
       <string>MD5</string>
      </property>
     </widget>
    </item>
    <item row="9" column="1">
     <widget class="QLabel" name="md5content">
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item row="10" column="0">
     <widget class="Line" name="line_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="pushButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>340</y>
     <width>83</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>1. Be Alice</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_2">
   <property name="geometry">
    <rect>
     <x>100</x>
     <y>340</y>
     <width>111</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>3. Read as Bob</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_3">
   <property name="geometry">
    <rect>
     <x>220</x>
     <y>340</y>
     <width>161</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>4. Be Bob, make secret</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_4">
   <property name="geometry">
    <rect>
     <x>390</x>
     <y>340</y>
     <width>201</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>5. Read as Alice, make secret</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_5">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>135</y>
     <width>83</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>8</pointsize>
    </font>
   </property>
   <property name="text">
    <string>random</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_6">
   <property name="geometry">
    <rect>
     <x>510</x>
     <y>160</y>
     <width>83</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>8</pointsize>
    </font>
   </property>
   <property name="text">
    <string>random</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_7">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>400</y>
     <width>83</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>Encrypt</string>
   </property>
  </widget>
  <widget class="QPushButton" name="pushButton_8">
   <property name="geometry">
    <rect>
     <x>150</x>
     <y>400</y>
     <width>83</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>Decrypt</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>encapp</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>encapp</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>encapp</receiver>
   <slot>runAlice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>191</x>
     <y>472</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>encapp</receiver>
   <slot>runBob()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>393</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>encapp</receiver>
   <slot>readBob()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>355</y>
    </hint>
    <hint type="destinationlabel">
     <x>148</x>
     <y>407</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>encapp</receiver>
   <slot>readAlice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>431</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>408</x>
     <y>418</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sharedSecretLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>encapp</receiver>
   <slot>updateMD5()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>173</x>
     <y>215</y>
    </hint>
    <hint type="destinationlabel">
     <x>202</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>encapp</receiver>
   <slot>randomA()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>529</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>524</x>
     <y>267</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>encapp</receiver>
   <slot>randomB()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>548</x>
     <y>176</y>
    </hint>
    <hint type="destinationlabel">
     <x>554</x>
     <y>275</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_7</sender>
   <signal>clicked()</signal>
   <receiver>encapp</receiver>
   <slot>encrypt()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>61</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>22</x>
     <y>414</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_8</sender>
   <signal>clicked()</signal>
   <receiver>encapp</receiver>
   <slot>decrypt()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>417</y>
    </hint>
    <hint type="destinationlabel">
     <x>92</x>
     <y>442</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>runAlice()</slot>
  <slot>runBob()</slot>
  <slot>readBob()</slot>
  <slot>readAlice()</slot>
  <slot>updateMD5()</slot>
  <slot>randomA()</slot>
  <slot>randomB()</slot>
  <slot>encrypt()</slot>
  <slot>decrypt()</slot>
 </slots>
</ui>
